### PR TITLE
graph pinned message nodejs sample

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

File: samples/graph-pinned-messages/nodejs/controller/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const teamTagController = require('./pin-message-controller');
+const express = require('express');
+const router = express.Router();
+
+// Routes for the API calls.
+router.get('/getGraphAccessToken', teamTagController.getGraphAccessToken);
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/controller/index.js:9]: teamTagController ??

File: samples/graph-pinned-messages/nodejs/controller/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+router.get('/pinMessage', teamTagController.pinNewMessage);
+router.get('/unpinMessage', teamTagController.unpinMessage);
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/controller/index.js:11]: Check these as well

File: samples/graph-pinned-messages/nodejs/controller/pin-message-controller.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    recentMessages.value.map(message => {
+        if(message.messageType == "message") {
+            var messageDetails = {
+                id: message.id,
+                value: message.body.content.replace(/<[^>]+>/g, '')
+            }
+    
+            messageList.push(messageDetails);
+        }
+
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/controller/pin-message-controller.js:55]: Remove extra space

File: samples/graph-pinned-messages/nodejs/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "grapgpinnedmessage-nodejs",
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/package.json:2]: grapg =? graph

File: samples/graph-pinned-messages/nodejs/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "axios": "^0.21.1",
+    "body-parser": "^1.19.0",
+    "cors": "^2.8.5",
+    "dotenv": "^8.2.0",
+    "ejs": "^3.1.6",
+    "express": "^4.17.1",
+    "isomorphic-fetch": "^3.0.0",
+    "node-fetch": "^2.6.1",
+    "nodemon": "^2.0.7",
+    "parallel-each": "^1.1.7",
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/package.json:26]: Are we using these packages ?

File: samples/graph-pinned-messages/nodejs/teamsAppManifest/manifest.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.12/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.12",
```
Comment (ChetanSharma-msft) [inline samples/graph-pinned-messages/nodejs/teamsAppManifest/manifest.json:3]: Did we deploy this sample?

image

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Reviewed and added my comments.

File: samples/graph-pinned-messages/nodejs/ClientApp/src/components/configure.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import * as microsoftTeams from "@microsoft/teams-js";
+
+// Configure page.
+const Configure = props => {
+    useEffect(() => {
+        microsoftTeams.app.initialize().then(() => {
+            microsoftTeams.app.notifySuccess();
+            microsoftTeams.pages.config.registerOnSaveHandler(function (saveEvent) {
+                microsoftTeams.pages.config.setConfig({
+                    entityID: "TeamworkTagsTab",
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/graph-pinned-messages/nodejs/ClientApp/src/components/configure.jsx]: Changed the entityId  and yes dashboard page is required

File: samples/graph-pinned-messages/nodejs/ClientApp/src/index.css [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+.details:hover, .details-sidepanel:hover {
+ box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
+}
+
+.add-card-button {
+  width: 21vw;
+  height: 32px;
+  cursor: pointer;
+  margin-top: 10px;
+  background-color: #484644;
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/graph-pinned-messages/nodejs/ClientApp/src/index.css:148]: Yes it is default css for the initial page of client app

File: samples/graph-pinned-messages/nodejs/package.json (hunk 5 above)
Comment (Prithvi-MSFT) [reply samples/graph-pinned-messages/nodejs/package.json]: Yes apart from parallel-each.. these packages are required

File: samples/graph-pinned-messages/nodejs/README.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+## Key features
+
+1. Pin new message in chat.
+
+![Pinned message](Images/PinMessage.png)
+
+2. The pinned message will be shown in tab.
+
+![Tab page](Images/TabImage.png)
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/graph-pinned-messages/nodejs/README.md:27]: Yes images are displayed fine

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**